src/cc_mm.c

```diff
@@ -82,6 +82,22 @@ _cc_realloc(void *ptr, size_t size, const char *name, int line)
     return p;
 }
 
+void *
+_cc_realloc_test(void *ptr, size_t size, const char *name, int line)
+{
+    void *p = NULL, *pr;
+
+    if ((pr = realloc(ptr, size)) == NULL || (p = malloc(size)) == NULL) {
```

Comment:
Style wise, I prefer breaking down a compound expression like this, and add some comments about what we are trying to achieve.
```C
pr = realloc(ptr, size); / * use the native realloc to make sure we copy the right amount of data to the location pointed to by pr */
p = malloc(size); /* allocate at a new address (and memcpy later) to make sure cc_realloc returns a different address than ptr */ 
if (pr == NULL || p == NULL) {
...
}
```